gax/status.cc

```diff
@@ -69,5 +69,9 @@ std::ostream& operator<<(std::ostream& os, Status const& rhs) {
   return os << rhs.message() << " [" << rhs.code() << "]";
 }
 
+Status GrpcStatusToGaxStatus(grpc::Status s) {
```

Comment:
Were you planning to rename this / move it into the class (as per comment on #33)?

Reply:
Changing it to an alternative constructor for `gax::Status`.

Reply:
The GCS client library does not depend on gRPC at all today (the APIs are only REST+JSON), and thus we have avoided introducing gRPC dependencies in `google::cloud::Foo`.  I am not saying that you should do the same (I understand you are explicitly targeting gRPC-only for C++ GAPIC (if that is the correct name)). But adding the depending on gRPC for gax:: would make it a harder decision for me to use it in the GCS client library.

Reply:
Moving it back from being a constructor.

I think we can separate the concerns of gRPC agnosticism from actively supporting additional transports. Currently there's only the one library for gax, one for the generated client, and one for the generated client stub. If we were to split those into e.g. `gax-base` and `gax-grpc`, that might prevent unnecessary gRPC dependence.

Reply:
Small followup, hit comment too early.

I don't think the above proposed split would be too onerous, but I would argue it needs a little more thought and design, and it falls outside the scope of this review. I'm okay with with filing an issue for it, though.

Reply:
Opening an issue is a good idea. I did not mean to ask for major changes in this PR.

Reply:
+1 on filing an issue to follow up on this. If we can enable GCS to share our gax library (via gax-base or some other mechanism) without taking a dependency on grpc, then we definitely should do so.

Reply:
Tracked by #37